lib/Optimizer/Transforms/QuakeAddMetadata.cpp

```diff
@@ -98,6 +98,8 @@ struct QuakeFunctionAnalysis {
         auto allocValue = storeOp.getOperand(1);
         if (auto cp = allocValue.getDefiningOp<cudaq::cc::ComputePtrOp>())
           allocValue = cp.getBase();
+        if (auto castOp = allocValue.getDefiningOp<cudaq::cc::CastOp>())
+          allocValue = castOp.getOperand();
```

Comment:
Should this be a loop to peel off any number of casts? Maybe before/after a compute pointer?

Reply:
As you point out below, the downstream code below is looking for specific trees of IR nodes, so peeling off multiple casts here would require updating the code below, too, right?

Reply:
Likely, yes.